components/scream/src/share/grid/remap/coarsening_remapper.cpp

```diff
@@ -645,8 +663,19 @@ get_my_triplets_gids (const std::string& map_file,
   scorpio::set_decomp(map_file);
   scorpio::grid_read_data_array(map_file,"col",-1,cols.data(),cols.size());
   scorpio::eam_pio_closefile(map_file);
+
+  // Offset the cols ids to match the source grid.
+  // Determine the min id among the cols array, we
+  // also add the min_dof for the grid.
+  int remap_min_dof = cols[0];
+  for (int id=1; id<nlweights; id++) {
+    remap_min_dof = std::min(cols[id],remap_min_dof);
+  }
+  int global_remap_min_dof;
+  m_comm.all_reduce(&remap_min_dof,&global_remap_min_dof,1,MPI_MIN);
+
   for (auto& id : cols) {
-    --id; // Subtract 1 to get 0-based indices
+    id = id - global_remap_min_dof + src_grid->get_global_min_dof_gid(); 
```

Comment:
This is fine, but I think that `-=` is more cplusplus-y than `id = id - ...`. However, given that you do - and +, perhaps precompute the offset for legibility.
```
int offset = global_remap_min_dof + src_grid->get_global_min_dof_gid();
for (auto& id : cols) {
  id -= offset;
}
```